src/templates/Dockerfile

```diff
@@ -27,14 +27,16 @@ COPY src/templates/addrman.patch /tmp/
 
 
 # Clone and patch and build stage
-FROM deps as build
+FROM deps AS build
 ENV BITCOIN_PREFIX=/opt/bitcoin
 WORKDIR /build
 
 RUN set -ex \
     && cd /build \
-    && git clone --depth 1 --branch "${BRANCH}" "https://github.com/${REPO}" \
+    && git clone --depth 1 "https://github.com/${REPO}" \
     && cd bitcoin \
+    && git fetch --depth 1 origin "$COMMIT_SHA" \
+    && git checkout "$COMMIT_SHA" \
```

Comment:
TIL you can do this: fetch a commit from another branch even though you cloned depth 1

Reply:
`fetch` is _from the remote_, so really the `clone` is just needed to setup `git` with the remote, then you can `fetch` any commit_sha (at depth 1) after that.

It's prob even possible to do in a one-liner, but I don't got time for that